tests/unit/task_processor/test_unit_task_processor_processor.py

```diff
@@ -91,6 +175,43 @@ def _dummy_recurring_task():
     assert task_run.error_details is None
 
 
+def test_run_recurring_tasks_runs_locked_task_after_tiemout(
+    db: None,
+    monkeypatch: MonkeyPatch,
+) -> None:
+    # Given
+    monkeypatch.setenv("RUN_BY_PROCESSOR", "True")
+
+    @register_recurring_task(run_every=timedelta(hours=1))
+    def _dummy_recurring_task():
+        cache.set(DEFAULT_CACHE_KEY, DEFAULT_CACHE_VALUE)
+
+    task = RecurringTask.objects.get(
+        task_identifier=_dummy_recurring_task.task_identifier
+    )
+    task.is_locked = True
+    task.locked_at = timezone.now() - timedelta(hours=1)
+    task.save()
+
+    # When
+    task_runs = run_recurring_tasks()
+
+    # Then
+    assert cache.get(DEFAULT_CACHE_KEY)
```

Comment:
```suggestion
    assert cache.get(DEFAULT_CACHE_KEY) == DEFAULT_CACHE_VALUE
```

Reply:
I do also think it's worth adding an assert to the beginning of the test that `DEFAULT_CACHE_KEY` doesn't already exist in the cache, or a `cache.delete(DEFAULT_CACHE_KEY)` perhaps?

Reply:
Done